### ISSUE TITLE
Cronjob for IMAP fetch not working with special chars in password

### ISSUE THREAD

**Issue by sstidl** [sameersbn/docker-redmine#457]
Using image: sameersbn/redmine:latest as of today
REDMINE_VERSION=4.1.1

Crontab of redmine user looks like this:
```
root@xxxxx:/home/redmine/redmine# crontab -l -u redmine
REDMINE_USER=redmine
REDMINE_INSTALL_DIR=/home/redmine/redmine
REDMINE_DATA_DIR=/home/redmine/data
REDMINE_RUNTIME_ASSETS_DIR=/etc/docker-redmine/runtime
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
TZ=Europe/Vienna
*/1 * * * * cd /home/redmine/redmine && bundle exec rake redmine:email:receive_imap host=mail.example.org port=993 ssl=true  username=redmine@example.org password=asdfasdfasdf%12                            unknown_user=ignore         no_permission_check=0         no_account_notice=true                  project=default         project_from_subaddress=redmine@example.org                  tracker=Task                                     RAILS_ENV=production >> /var/log/redmine/redmine/cron_rake.log 2>&1
@hourly cd /home/redmine/redmine && ./bin/rails runner "Repository.fetch_changesets" -e production >> /var/log/redmine/redmine/cron_rake.log 2>&1
00 04 * * * /usr/bin/redmine-backup-create >> /var/log/redmine/redmine/backups.log 2>&1
```

Problem is: Cronjob is never executed (no logs in /var/log/redmine/redmine/cron_rake.log or on the IMAP-Server)

I changed the password to something without % in it => works again.

Seems, the shell or cron doesnt like special chars.

**Comment by sstidl** [comment]
didnt find any errors in logs, so was difficult to find the cause

**Comment by frank-dspeed** [comment]
@sstidl how should we address that your using linux software you need to do char escaping should we really sanitize all your files for you? we do not even know which are modifyed by you or not?

**Comment by jcormier** [comment]
@frank-dspeed The email password field is a docker env variable that we provide.  At the very least this is a documentation issue as the user isn't likely to know this gets passed to a bash cron script.

It seems to me that we might want to at least quote the password="asdfasdfasdf%12" field when we generate the cronjob

**Comment by jcormier** [comment]
Code is located here: https://github.com/sameersbn/docker-redmine/blob/29f9a1d13a9deb93152975a9395f13d52fb197f2/assets/runtime/functions#L453

Though need to figure out if it's cron/bash that doesn't like the % or the redmine:email:receive_imap task.

Since as far as I'm aware, % is not a special character in either bash or cron.

**Comment by jcormier** [comment]
Looks like cron does treat % characters as special characters.
https://serverfault.com/a/274480/62690

**Comment by sstidl** [comment]
> Looks like cron does treat % characters as special characters.
> https://serverfault.com/a/274480/62690

That's why I was wondering why it didn't work


@frank-dspeed I spent some time to find out where the crontab was and as I stated, there is no error message anywhere in a logfile.

I also tried to quote the environment variable in docker-compose.yml without any success.

My workaround was to change the password to something without the %

If you need more input, as labeled, tell me what you need

**Comment by jcormier** [comment]
@sstidl This seems to be an issue that even redmine's documentation is ignoring, as they make no mention of requiring certain escapes.  Though at least it's obvious that cron is being used.
https://www.redmine.org/projects/redmine/wiki/RedmineReceivingEmails#Fetching-emails-from-an-IMAP-server

I was able to run the rake job with a % password successfully by escaping the % with a \

I think in the long run, my preference would be to switch to using the rufus scheduler to run these tasks so we don't have to worry about escaping characters for both cron and bash.
https://www.redmine.org/projects/redmine/wiki/RedmineReceivingEmails#Schedule-email-receiving-with-Rufus-Scheduler

But I don't have time to do this right now, so if escaping the % works for now, then I'll just update the documentation with a warning.

**Comment by frank-dspeed** [comment]
@jgillich if we would want to we could maybe generate a escape function that simply escapes all special chars that we know? but that would then be out of my view a standalone shared project that could be reused so chances are high that some one did a chare escape tool already.

**Comment by frank-dspeed** [comment]
This got Closed by PR: #457 if you do not feel so please reopen.
